### ISSUE TITLE
Code and text feedback

### ISSUE THREAD

**Issue by lassedochreden** [BodenmillerGroup/IMCDataAnalysis#15]
Hi @nilseling, 

as we have discussed before I wanted to open this issue for minor code/text updates in your analysis workflow. I am considering to continuously add to this issue, if I encounter something. Hope that is fine with you. 

For now, I have following three (minor) points: 

- [x] *7.3 Image-level quality control* (also applicable for 7.4 Cell-level quality control): 

You could consider reordering image area covered by cells (number of cells per image divided by the image size) from low to high and add horizontal lines for median/quantiles:
```
image_cell_area <- colData(spe) %>%
    as.data.frame() %>%
    group_by(sample_id) %>%
    summarize(cell_area = sum(area),
           no_pixels = mean(width_px) * mean(height_px)) %>%
    mutate(covered_area = cell_area / no_pixels)

 ggplot(image_cell_area) +
  geom_point(aes(reorder(sample_id,covered_area), covered_area, color = covered_area))+ 
  ylim(c(0, 1))+
  geom_hline(yintercept = median(cell_density$covered_area), linetype="dashed", color = "black")+
  geom_hline(yintercept = quantile(cell_density$covered_area)[2],linetype="dashed", color = "grey")+
  geom_hline(yintercept = quantile(cell_density$covered_area)[4],linetype="dashed", color = "grey")+
  theme_classic(base_size = 15)+
  theme(axis.text.x = element_text(angle = 90, hjust = 1, size = 8))+
  scale_color_continuous(type = "viridis")+
  ylab("% covered area per ROI")+xlab("")
```

- [x] *7.3 Image-level quality control*: 

You calculate the `SNR` here for each image *separately*. It may be worthwhile to discuss that one could also do that over all images *simultaneously*, if the count distributions for a given marker would align.

- [x] *11. Single-cell visualization* 

You state that you will use the `dittoSeq` package here. If you consider to also aggregate your cells across specific categories (e.g. clusters), I would personally use the `ComplexHeatmap` package (dittoSeq also has an option to set `complex = TRUE`). I am happy to contribute code here if you would like. :)

**Comment by nilseling** [comment]
Hi @lassedochreden, thanks for the feedback! I'm only now at a point to go through the issues. I'd be super happy if you contribute a section on single-cell visualisation (assuming you'll keep it simple enough ;) ). You might also want to check out what [CATALYST](https://bioconductor.org/packages/release/bioc/vignettes/CATALYST/inst/doc/differential.html#visualization) has already implemented.

**Comment by lassedochreden** [comment]
Hi @nilseling, 
thanks for the response! Let's discuss the potential contribution a bit more soon.

Furthermore, here are some more points for consideration: 

- [x] **Section 6.1** Reminder that the spillover-matrix visualization is cut at Yb171
- [x] **7.3** “Finally, it can be beneficial to visualize the mean marker expression per image to identify images with outlying cell type compositions.” Not sure whether that fits well here - better to call it outlying marker expressions, since you did not perform any cell phenotyping / cell type designation at this point. 
- [x] **9.2.1** While you mention this for 9.2.2, one could also consider to mention for Rphenograph that it is advisable to test different numbers for k
- [x] **9.2.2** Order _average silhouette width plot_ from highest to lowest (see above).
- [x] **9.3** Call it _Classification approaches_ (not _cell type classification_ - I thought it was a bit misleading?)
- [x] **11.4** - Your note that one needs to test k for kmeans for cn assignment only comes after the full section. Maybe you could add this note earlier in the text? - summarize(count = n()) could be replaced by count()? 

- [x] **General:** 
Mention _iSEE_ package for interactive visualization of spe/sce objects? - probably in Section 5
kmeans (set.seed vs. nstart?)

Looking forward to discuss more :)

**Comment by lassedochreden** [comment]
Two more small things - to keep you busy ;) 

- [ ] Section 8 (but applicable in general): `multi_dittoDimPlot` does not create a legend for expression values - maybe use `plotReducedDim`? 

- [x] Different color scheme for sample_id (related to the patient_id colors) would be nice (e.g. Patient 1 ~ red - Samples from Patient 1 ~ reddish)

**Comment by nilseling** [comment]
> Hi @nilseling,
> 
> as we have discussed before I wanted to open this issue for minor code/text updates in your analysis workflow. I am considering to continuously add to this issue, if I encounter something. Hope that is fine with you.
> 
> For now, I have following three (minor) points:
> 
> * [x]  _7.3 Image-level quality control_ (also applicable for 7.4 Cell-level quality control):
> 
> You could consider reordering image area covered by cells (number of cells per image divided by the image size) from low to high and add horizontal lines for median/quantiles:
> 
> ```
> image_cell_area <- colData(spe) %>%
>     as.data.frame() %>%
>     group_by(sample_id) %>%
>     summarize(cell_area = sum(area),
>            no_pixels = mean(width_px) * mean(height_px)) %>%
>     mutate(covered_area = cell_area / no_pixels)
> 
>  ggplot(image_cell_area) +
>   geom_point(aes(reorder(sample_id,covered_area), covered_area, color = covered_area))+ 
>   ylim(c(0, 1))+
>   geom_hline(yintercept = median(cell_density$covered_area), linetype="dashed", color = "black")+
>   geom_hline(yintercept = quantile(cell_density$covered_area)[2],linetype="dashed", color = "grey")+
>   geom_hline(yintercept = quantile(cell_density$covered_area)[4],linetype="dashed", color = "grey")+
>   theme_classic(base_size = 15)+
>   theme(axis.text.x = element_text(angle = 90, hjust = 1, size = 8))+
>   scale_color_continuous(type = "viridis")+
>   ylab("% covered area per ROI")+xlab("")
> ```
> 

Done. But I would not plot the median line.

> * [ ]  _7.3 Image-level quality control_:
> 
> You calculate the `SNR` here for each image _separately_. It may be worthwhile to discuss that one could also do that over all images _simultaneously_, if the count distributions for a given marker would align.
> 

I'm using an Otso thresholding here which only really works on a per-image basis. I will therefore leave as is.

> * [x]  _11. Single-cell visualization_
> 
> You state that you will use the `dittoSeq` package here. If you consider to also aggregate your cells across specific categories (e.g. clusters), I would personally use the `ComplexHeatmap` package (dittoSeq also has an option to set `complex = TRUE`). I am happy to contribute code here if you would like. :)

Done by you :)

**Comment by nilseling** [comment]
I think all of these comments have been addressed in #43. Let's open a new issue when we see missing pieces.
